#! /usr/bin/env python

PACKAGE='jsk_perception'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gaussian_blur_size", int_t, 0, "the size of gaussian blur (should be odd number)", 0, 9, 30)
gen.add("gaussian_sigma_x", double_t, 0, "sigma x of gaussian kernel", 0, 2, 10)
gen.add("gaussian_sigma_y", double_t, 0, "sigma y of gaussian kernel", 0, 2, 10)
gen.add("edge_threshold", double_t, 0, "edge threshold", 0, 200, 1000)
gen.add("voting_threshold", double_t, 0, "voting threshold", 0, 100, 1000)
gen.add("dp", int_t, 0, "dp", 0, 2, 10)
gen.add("min_circle_radius", int_t, 0, "the minimum size of the circle", 0, 0, 500)
gen.add("max_circle_radius", int_t, 0, "the maximum size of the circle", 0, 0, 2000)
exit(gen.generate(PACKAGE, "hough_cirlces", "HoughCircles"))
